src/main/java/com/google/devtools/build/lib/syntax/UnaryOperatorExpression.java

```diff
@@ -63,29 +63,31 @@ private static Object evaluate(TokenKind op, Object value, Location loc)
         return !EvalUtils.toBoolean(value);
 
       case MINUS:
-        if (!(value instanceof Integer)) {
-          throw new EvalException(
-              loc,
-              String.format(
-                  "unsupported operand type for -: '%s'", EvalUtils.getDataTypeName(value)));
+        if (value instanceof Integer) {
+          try {
+            return Math.negateExact((Integer) value);
+          } catch (ArithmeticException e) {
+            // Fails for -MIN_INT.
+            throw new EvalException(loc, e.getMessage());
+          }
         }
-        try {
-          return Math.negateExact((Integer) value);
-        } catch (ArithmeticException e) {
-          // Fails for -MIN_INT.
-          throw new EvalException(loc, e.getMessage());
+
+      case PLUS:
+        if (value instanceof Integer) {
+          return value;
         }
+
       case TILDE:
-        if (!(value instanceof Integer)) {
-          throw new EvalException(
-              loc,
-              String.format(
-                  "unsupported operand type for ~: '%s'", EvalUtils.getDataTypeName(value)));
+        if (value instanceof Integer) {
+          return ~((Integer) value);
         }
-        return ~((Integer) value);
 
       default:
-        throw new AssertionError("Unsupported unary operator: " + op);
+        throw new EvalException(
```

Comment:
The default should be just a break, and throw statement should be outside the switch so that it is reached if any case doesn't return.